tests/user_examples/extra_src/shared/wetting_coupled_spatial_temporal_method.h

```diff
@@ -0,0 +1,104 @@
+ * ------------------------------------------------------------------------- *
+ * SPHinXsys (pronunciation: s'finksis) is an acronym from Smoothed Particle *
+ * Hydrodynamics for industrial compleX systems. It provides C++ APIs for    *
+ * physical accurate simulation and aims to model coupled industrial dynamic *
+ * systems including fluid, solid, multi-body dynamics and beyond with SPH   *
+ * (smoothed particle hydrodynamics), a meshless computational method using  *
+ * particle discretization.                                                  *
+ *                                                                           *
+ * SPHinXsys is partially funded by German Research Foundation               *
+ * (Deutsche Forschungsgemeinschaft) DFG HU1527/6-1, HU1527/10-1,            *
+ *  HU1527/12-1 and HU1527/12-4.                                             *
+ *                                                                           *
+ * Portions copyright (c) 2017-2023 Technical University of Munich and       *
+ * the authors' affiliations.                                                *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain a *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.        *
+ *                                                                           *
+ * ------------------------------------------------------------------------- */
+/**
+ * @file 	fluid_surface_complex.h
+ * @brief 	Here, we define the algorithm classes for fluid dynamics within the body.
+ * @details We consider here weakly compressible fluids. The algorithms may be
+ * 			different for free surface flow and the one without free surface.
+ * @author	Chi Zhang and Xiangyu Hu
+ */
+
+#ifndef WETTING_COUPLED_SPATIAL_TEMPORAL_COMPLEX_H
+#define WETTING_COUPLED_SPATIAL_TEMPORAL_COMPLEX_H
+
+#include "fluid_surface_complex.h"
+
+namespace SPH
+{
+namespace fluid_dynamics
+{
+/**
+ * @class WettingCoupledFreeSurfaceIndicationComplex
+ * @brief indicate the interfacial fluid particles near the fluid-structure intersurface.
+ */
+class WettingCoupledFreeSurfaceIndicationComplex : public FreeSurfaceIndicationComplex
+{
+  public:
+    WettingCoupledFreeSurfaceIndicationComplex(BaseInnerRelation &inner_relation,
+                                               BaseContactRelation &contact_relation, Real threshold = 0.75, Real criterion = 0.99);
+    explicit WettingCoupledFreeSurfaceIndicationComplex(ComplexRelation &complex_relation, Real threshold = 0.75, Real criterion = 0.99);
+    virtual ~WettingCoupledFreeSurfaceIndicationComplex(){};
+
+    inline void interaction(size_t index_i, Real dt = 0.0)
+    {
+        FreeSurfaceIndicationInner::interaction(index_i, dt);
+
```

Comment:
" WettingCoupledFreeSurfaceIndicationComplex" is derived from " FreeSurfaceIndicationComplex", where here you call " FreeSurfaceIndicationInner::interaction(index_i, dt)"  and then add the contribution from contact, viz. herein wall.  
Seems that  " WettingCoupledFreeSurfaceIndicationComplex" derived from " FreeSurfaceIndicationComplex" is a confusing operation, would you please clarify the underlying principle?